src/catch2/internal/catch_test_macro_impl.hpp

```diff
@@ -108,7 +114,10 @@
         Catch::AssertionHandler catchAssertionHandler( macroName##_catch_sr, CATCH_INTERNAL_LINEINFO, CATCH_INTERNAL_STRINGIFY(expr) ", " CATCH_INTERNAL_STRINGIFY(exceptionType), resultDisposition ); \
         if( catchAssertionHandler.allowThrows() ) \
             try { \
+                CATCH_INTERNAL_START_WARNINGS_SUPPRESSION \
+                CATCH_INTERNAL_SUPPRESS_USELESS_CAST_WARNINGS \
                 static_cast<void>(expr); \
```

Comment:
Any reason this isn't a variadic macro?

Reply:
Legacy, it was unchanged since Catch v1, which supported C++98 compilers (which do not guarantee variadic macros).

It should be fine to change it to a variadic macro.

Reply:
However, upon closer inspection, I don't think this will change anything. The user-facing macro is IIRC `THROWS_AS(expr, exceptionType)`, which cannot be made meaningfully variadic either way, so there would be no benefit to changing this here.